dfaker.py

```diff
@@ -0,0 +1,387 @@
+	for row in solution:
+		bolus_or_wizard = random.randint(0, 5)
+		if bolus_or_wizard == 2 or bolus_or_wizard == 4:
+			bolus_events.append(row)
+		else:
+			wizard_events.append(row)
+	return bolus_events, wizard_events
+
+def convert_to_mmol(iterable):
+	conversion_factor = 18.01559
+	return [reading / conversion_factor for reading in iterable]
+
+def make_timesteps(start_timestamp, timelist):
+	""" Convert list of floats representing time into epoch time"""
+	timesteps = []
+	epoch_ts = start_timestamp.strftime('%s')
+	for time_item in timelist:
+		new_time = int(epoch_ts) + int(time_item * 60) 
+		timesteps.append(new_time)
+	return timesteps
+
+def cbg(gluc, timesteps, params):
+	for value, timestamp in zip(convert_to_mmol(gluc), timesteps):
+		cbg_reading = {}
+		cbg_reading["deviceId"] = "MiniMed 530G - 551-=-53875424"
+		cbg_reading["id"] = str(uuid.uuid4())
+		cbg_reading["uploadId"] = "upid_fdbde582fe2b"
+		cbg_reading["type"] = "cbg"
+		cbg_reading["value"] = value
+		cbg_reading["deviceTime"] = time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime(timestamp))
+		cbg_reading["timezoneOffset"] = get_offset(params)
+		cbg_reading["time"] = (time.strftime('%Y-%m-%dT%H:%M:%S.000Z', time.gmtime(timestamp 
+								- cbg_reading["timezoneOffset"]*60)))
+		cbg_reading["units"] = "mmol/L"
+		dfaker.append(cbg_reading)
+
+def smbg(gluc, timesteps, params):
+	fingersticks_per_day = params['smbg_freq']
+	total = (24 * 60) / 5
+	increment = int(total / fingersticks_per_day)
+	time_gluc = list(zip(convert_to_mmol(gluc), timesteps))
+	for value, timestamp in time_gluc[::increment]:
+		smbg_reading = {}
+		smbg_reading["deviceId"] = "MiniMed 530G - 551-=-53875424"
+		smbg_reading["id"] = str(uuid.uuid4())
+		smbg_reading["type"] = "smbg"
+		smbg_reading["uploadId"] = "upid_fdbde582fe2b"
+		smbg_reading["value"] = value + random.uniform(-1.5, 1.5) #in mmol/L
+		randimize_time = random.randint(-6000, 6000)
+		smbg_reading["deviceTime"] = (time.strftime('%Y-%m-%dT%H:%M:%S', time.gmtime(timestamp 
+									 + randimize_time)))
+		smbg_reading["timezoneOffset"] = get_offset(params) 
+		smbg_reading["time"] = (time.strftime('%Y-%m-%dT%H:%M:%S.000Z', time.gmtime(timestamp                          
+								+ randimize_time - smbg_reading["timezoneOffset"]*60)))
+		smbg_reading["units"] = "mmol/L"
+		dfaker.append(smbg_reading)
+
+def bolus(carbs, timesteps, params):
+	for value, timestamp in zip(carbs, timesteps):		
+		normal_or_square = random.randint(0, 4)
+		if normal_or_square == 3:
+			dual_square_bolus(value, timestamp, params)
+		else:
+			normal_bolus(value, timestamp, params)
+
+def dual_square_bolus(value, timestamp, params):
+	bolus_entry = {}
+	bolus_entry["deviceId"] = "MiniMed 530G - 551-=-53875424"
+	bolus_entry["id"] = str(uuid.uuid4())
+	bolus_entry["uploadId"] = "upid_fdbde582fe2b"
+	bolus_entry["type"] = "bolus"
+	bolus_entry["subType"] = "dual/square"
+	bolus_entry["normal"] = random.uniform((value / 10) / 3, (value / 10) / 2) 
```

Comment:
Explain me your logic on choosing random values using these parameters - `(value / 10) / 3` and `(value / 10) / 2`? It's not immediately obvious to me.

Related:
- This is my bad, probably, because I remembering using the word "override" to describe how people don't always use the recommended bolus value. You can override by taking more or less insulin than the recommendation and both should have an equal distribution in the demo data.
- Let's round the bolus values (as well as recommendations in the wizard) to 2 or 3 decimal places. Ideally rounded to .005 if you want that challenge. (Most pumps don't have a finer-grained control than that.)